'nrf_cache_data_unit_validity_check' declared 'static' but never defined

### Describe the bug

There is a ifdef misalignment around `NRF_CACHE_HAS_ANY_CACHEINFO` in nrf_cache.h which results in a build warning.


### Regression

- [ ] This is a regression.

### Steps to reproduce

1. mkdir build && cd build
2. cmake -GNinja -DBOARD=nrf54l15dk/nrf54l15/cpuapp/ns ../tests/kernel/cache/ 
3. ninja
4. see warning


### Relevant log output

```shell
In file included from drivers/cache/cache_nrf.c:9:
modules/hal/nordic/nrfx/hal/nrf_cache.h:748:24: warning: 'nrf_cache_data_unit_validity_check' declared 'static' but never defined [-Wunused-function]
  748 | NRF_STATIC_INLINE bool nrf_cache_data_unit_validity_check(NRF_CACHE_Type const * p_reg,
      |                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

https://github.com/zephyrproject-rtos/zephyr/actions/runs/19997797171/job/57348087635#step:14:5476
16 failures in weekly:
https://github.com/zephyrproject-rtos/zephyr/actions/runs/19997797171

### Impact

Functional Limitation – Some features not working as expected, but system usable.

### Environment

- OS any,
- Zephyr SDK 0.17.4
- main as of now 923ece1eea7d303d06b617bb85159832c6b9c12e

### Additional Context

_No response_

--- a/west.yml
+++ b/west.yml
@@ -200,7 +200,7 @@ manifest:
       groups:
         - hal
     - name: hal_nordic
-      revision: 248eadcacf976bbd27f1c0bc0dd3f11d8ec8657e
+      revision: a83db66acbeca0bfef157a0c3482c07ddbb82555
       path: modules/hal/nordic
       groups:
         - hal

